streamer/src/nonblocking/quic.rs

```diff
@@ -1083,27 +1103,40 @@ async fn handle_connection(
                 sleep(throttle_duration).await;
             }
         }
-        stream_load_ema.increment_load(params.peer_type);
+        stream_load_ema.increment_load(peer_type);
         stream_counter.stream_count.fetch_add(1, Ordering::Relaxed);
         stats.total_streams.fetch_add(1, Ordering::Relaxed);
         stats.total_new_streams.fetch_add(1, Ordering::Relaxed);
-        let packet_sender = &params.packet_sender;
-        let mut maybe_batch = None;
+
+        let mut meta = Meta::default();
+        meta.set_socket_addr(&remote_addr);
+        meta.set_from_staked_node(matches!(peer_type, ConnectionPeerType::Staked(_)));
+        let mut accum = PacketAccumulator::new(meta);
+
+        // Virtually all small transactions will fit in 1 chunk. Larger transactions will fit in 1
+        // or 2 chunks if the first chunk starts towards the end of a datagram. A small number of
+        // transaction will have other protocol frames inserted in the middle. Empirically it's been
+        // observed that 4 is the maximum number of chunks txs get split into.
+        //
+        // Bytes values are small, so overall the array takes only 128 bytes, and the "cost" of
+        // overallocating a few bytes is negligible compared to the cost of having to do multiple
+        // read_chunks() calls.
+        let mut chunks: [Bytes; 4] = array::from_fn(|_| Bytes::new());
```

Comment:
If 2 is predominant  case, does not make sense to use 2 instead of 4?

Reply:
Because - as the comment explains - allocating an extra 64 bytes on the stack is better than having to call read_chunks(2 chunks) twice in the worst case, since that executes a lot more code, takes a contended lock (the connection lock), etc.